packages/client/test/cli/cli.spec.ts

```diff
@@ -24,25 +51,145 @@ describe('[CLI]', () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = ['--rpcEngine', '--rpcEngineAuth=false', '--dev=poa']
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('http://')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('engine'), 'read from HTTP RPC')
+          assert.ok(message.includes('rpcEngineAuth=false'), 'read from HTTP RPC')
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  it('should start engine rpc on custom port', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = ['--rpcEngine', '--rpcEnginePort=8552', '--dev=poa']
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('http://')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('8552'), 'read from HTTP RPC')
+          child.kill()
+          resolve(undefined)
+        }
+        if (message.toLowerCase().includes('error')) {
+          child.kill(9)
+          assert.fail(`client encountered error: ${message}`)
+        }
+      })
+      child.stderr.on('data', (data) => {
+        const message: string = data.toString()
+        assert.fail(`stderr: ${message}`)
+      })
+      child.on('close', (code) => {
+        if (typeof code === 'number' && code > 0) {
+          assert.fail(`child process exited with code ${code}`)
+        }
+      })
+    })
+  }, 18000)
+  it('should start engine rpc on custom address', async () => {
+    const file = require.resolve('../../dist/bin/cli.js')
+    const cliArgs = ['--rpcEngine', '--rpcEngineAddr="0.0.0.0"', '--dev=poa']
+    const child = spawn(process.execPath, [file, ...cliArgs])
+    return new Promise((resolve) => {
+      child.stdout.on('data', async (data) => {
+        const message: string = data.toString()
+        if (message.includes('http://')) {
+          // if http endpoint startup message detected, call http endpoint with RPC method
+          assert.ok(message.includes('0.0.0.0'), 'read from HTTP RPC')
```

Comment:
As above, add an RPC call that verifies you get the engine authentication error message when calling an engine endpoint on the custom address and then clarify the test message

Reply:
fixed